spec/system/teams_page_system_test.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2008-2017, Puzzle ITC GmbH. This file is part of
+#  Cryptopus and licensed under the Affero General Public License version 3 or later.
+#  See the COPYING file at the top-level directory or at
+#  https://github.com/puzzle/cryptopus.
+
+require 'rails_helper'
+
+
+describe 'TeamsPage', type: :system, js: true do
+  include SystemHelpers
+
+  it 'clicks through teams page' do
+    login_as_user(:bob)
+
+    account1 = accounts(:account1)
+
+    visit('/teams')
+
+    # Search Account
+    expect(page).to have_selector('input#search')
+    find('input#search', visible: false).set account1.accountname
+    expect(page).to have_text(account1.accountname)
+
+    # Look if the Nesting is expanded
+    folder_expanded?
+    team_expanded?
+
+    # Click on Team in Sidebar
+    team1 = teams(:team1)
+    folder1 = folders(:folder1)
+
+    within('#sidebar') do
+      # Click on expand arrow in team card
+      find('a', text: team1.name, visible: false).click
+    end
+
+    expect(page).to have_text(folder1.name)
```

Comment:
Check for within the sidebar or for the teams page? Whether it is shown in sidebar is already done. So check this within the teams page. Is the team there expanded and the folder visible?

Reply:
Yes. You're right. I covered this already in the Teams page test.

Reply:
This line should check if in the teams page, the folder is present? But it looks in the whole page, so that it is always found in the sidebar, no matter if it is present in the teams page.

Reply:
I think you cannot achieve a function to return if something is visible or not. As compromise, I could make the Checks directly, without a function.